mmdet/models/losses/iou_loss.py

```diff
@@ -114,6 +116,125 @@ def giou_loss(pred, target, eps=1e-7):
     return loss
 
 
+@weighted_loss
+def diou_loss(pred, target, eps=1e-7):
+    """
+    Distance-IoU Loss: Faster and Better Learning for Bounding Box Regression
+    https://arxiv.org/abs/1911.08287
+
+    code refer to:
+    https://github.com/Zzh-tju/DIoU
+
+    Args:
+        pred (Tensor): Predicted bboxes of format (x1, y1, x2, y2),
+            shape (n, 4).
+        target (Tensor): Corresponding gt bboxes, shape (n, 4).
+        eps (float): Eps to avoid log(0).
+    Return:
+        Tensor: Loss tensor.
+    """
+    # overlap
+    lt = torch.max(pred[:, :2], target[:, :2])
+    rb = torch.min(pred[:, 2:], target[:, 2:])
+    wh = (rb - lt + 1).clamp(min=0)
```

Comment:
We need to take care about the `+1` when calculating the width, height and areas. Since MMDetection V2.0 uses new coordinate system, those +1 is unnecessary and may hurt performance in some datasets.

Reply:
got it!